instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/CustomOfflineOnlineSwitchTest.kt

```diff
@@ -0,0 +1,135 @@
+
+    override fun setupMockLocation(): Location {
+        return mockLocationUpdatesRule.generateLocationUpdate {
+            longitude = 13.361378213031003
+            latitude = 52.49813341962201
+        }
+    }
+
+    @Test
+    fun startNavigationOfflineThenSwitchToOnlineRouteWhenInternetAppears() = sdkTest(
+        timeout = INCREASED_TIMEOUT_BECAUSE_OF_REAL_ROUTING_TILES_USAGE
+    ) {
+        val originalTestRoute = setupBerlinEvRoute()
+        withMapboxNavigationAndOfflineTilesForRegion(
+            OfflineRegions.Berlin,
+            historyRecorderRule = mapboxHistoryTestRule
+        ) { navigation ->
+            OnlineRouteAlternativesSwitch().onAttached(navigation)
+            navigation.startTripSession()
+            stayOnPosition(
+                originalTestRoute.origin.latitude(),
+                originalTestRoute.origin.longitude(),
+                0.0f,
+            ) {
+                withoutInternet {
+                    val requestResult = navigation.requestRoutes(originalTestRoute.routeOptions)
+                        .getSuccessfulResultOrThrowException()
+                    assertEquals(RouterOrigin.Onboard, requestResult.routerOrigin)
+                    navigation.setNavigationRoutesAsync(requestResult.routes)
+                }
+                val onlineRoutes = navigation.routesUpdates().first {
+                    it.reason == RoutesExtra.ROUTES_UPDATE_REASON_NEW &&
+                        it.navigationRoutes.first().origin == RouterOrigin.Offboard
+                }
+                assertEquals(2, onlineRoutes.navigationRoutes.size)
+            }
+        }
+    }
+
+    @Test
+    fun startNavigationOfflineStayOfflineForAWhileThenTurnOnInternet() = sdkTest(
+        timeout = INCREASED_TIMEOUT_BECAUSE_OF_REAL_ROUTING_TILES_USAGE
+    ) {
+        val originalTestRoute = setupBerlinEvRoute()
+        withMapboxNavigationAndOfflineTilesForRegion(
+            OfflineRegions.Berlin,
+            historyRecorderRule = mapboxHistoryTestRule
+        ) { navigation ->
+            OnlineRouteAlternativesSwitch(
+                connectTimeoutMilliseconds = 100,
+                minimumRetryInterval = 500
+            ).onAttached(navigation)
+            navigation.startTripSession()
+            stayOnPosition(
+                originalTestRoute.origin.latitude(),
+                originalTestRoute.origin.longitude(),
+                0.0f,
+            ) {
+                withoutInternet {
+                    val requestResult = navigation.requestRoutes(originalTestRoute.routeOptions)
+                        .getSuccessfulResultOrThrowException()
+                    assertEquals(RouterOrigin.Onboard, requestResult.routerOrigin)
+                    navigation.setNavigationRoutesAsync(requestResult.routes)
+                    delay(4_000) // cause a few retry intervals to happen
+                }
+                val onlineRoutes = navigation.routesUpdates().first {
```

Comment:
You don't check here or in the test above that the retries are as frequent as we expect them to be. 
We can use `withTimeout` inside the test to limit only some part of it. In this case we might expect a new route within a few seconds.

Reply:
done